### PR TITLE
fix connection shutdown in arangosh

### PR DESCRIPTION
**Pull request by jsteemann**

### CODE REVIEW

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/3387

File: arangosh/Shell/V8ClientConnection.cpp [hunk 1]
```diff
@@ -158,16 +158,20 @@ void V8ClientConnection::setInterrupted(bool interrupted) {
   }
 }
 
-bool V8ClientConnection::isConnected() {
-  if (_connection) {
-    return _connection->state() == fuerte::Connection::State::Connected;
+bool V8ClientConnection::isConnected() const {
+  std::weak_ptr<fuerte::Connection> connection = _connection;
```
Comment (graetzer) [inline arangosh/Shell/V8ClientConnection.cpp]: wait why don't you just get the `_lock` ?
Also whyt about `requestDataRaw` and `requestData`

Comment (jsteemann) [reply arangosh/Shell/V8ClientConnection.cpp]: isConnected must not acquire `_lock`, as it seemed to be called under the lock already.

**Comment by mpoeter** [comment]
All accesses to `_connection` are now done via `std::atomic_*`... All accesses? No, there is still a lonely non-atomic write access in [line 101](https://github.com/arangodb/arangodb/pull/8375/files#diff-b862aa42dd9873e4cfea51e5d26296e4L101). :wink:

**Comment by jsteemann** [comment]
Thanks @mpoeter!

**Pull request merged**